### PR TITLE
Removed unnecessary variables and removed a dataleak bug

### PR DESCRIPTION
**Pull request by jaggbow**
This is the final version of dataloading. Removed all un-necessary variables and correctly defines the mean and variance for each subset

### CODE REVIEW

File: aiphysim/dataloading/dataloader_spacetime.py [hunk 1]
```diff
@@ -72,6 +66,8 @@ def __init__(
                 ],
                 axis=0,
             )
+            self.data = self.data.astype(np.float32)
```
Comment (sm7610) [inline aiphysim/dataloading/dataloader_spacetime.py:69]: What's the original data type here, and why do we need to convert it to float32?

**Review by sm7610 — CHANGES_REQUESTED** [review]

**Review by vict0rsch — APPROVED** [review]

File: aiphysim/losses.py [hunk 2]
```diff
@@ -133,6 +133,26 @@ def set_args(self, inputs, predictions, model):
         }
 
 
+class Unet3DLoss(BaseLoss):
+    def set_args(self, inputs, predictions, model):
+        """
+        Compute the weighted loss with respect to targets and predictions
+
+        Args:
```
[the remark this answers is not in the document]
Comment (jaggbow) [reply aiphysim/losses.py:141]: The last commit should solve this

[the remark this answers is not in the document]
Comment (sm7610) [reply aiphysim/losses.py:141]: You should also include a description for "model"

File: aiphysim/losses.py [hunk 3]
```diff
@@ -133,6 +133,19 @@ def set_args(self, inputs, predictions, model):
...
+            inputs (dict): dictionnary of target values
```
Comment (sm7610) [inline aiphysim/losses.py:142]: typo: dictionnary --> dictionary

File: aiphysim/losses.py [hunk 4]
```diff
@@ -133,6 +133,19 @@ def set_args(self, inputs, predictions, model):
...
+            predictions (dict): dictionnary of predicted values
```
Comment (sm7610) [inline aiphysim/losses.py:143]: typo: dictionnary --> dictionary

File: aiphysim/dataloading/dataloader_spacetime.py [hunk 5]
```diff
@@ -117,58 +101,44 @@ def __init__(
...
-                    hdata["tasks"]["w"][: int(nt * train_size)],
-                ],
-                axis=0,
-            )
+            self.data = self.data.transpose(1, 0, 3, 2)  # [t, c, z, x]
 
-            self._mean = np.mean(training_data, axis=(0, 2, 3))
-            self._std = np.std(training_data, axis=(0, 2, 3))
-
-        self.data = self.data.astype(np.float32)
```
Comment (sm7610) [inline aiphysim/dataloading/dataloader_spacetime.py:133]: Why are you changing the data type?

File: aiphysim/dataloading/dataloader_spacetime.py (hunk 1 above)
Comment (jaggbow) [reply aiphysim/dataloading/dataloader_spacetime.py]: The original type was float64 which is also known as "long". But the model requires a float32 as input, hence the type casting

File: aiphysim/dataloading/dataloader_spacetime.py (hunk 5 above)
Comment (jaggbow) [reply aiphysim/dataloading/dataloader_spacetime.py:133]: It was redundant, I already set it to float (which is float32 in pytorch) in the __getitem__

**Pull request merged**